modules/@apostrophecms/page/index.js

```diff
@@ -1404,6 +1412,17 @@ database.`);
         } catch (err) {
           return await self.serve500Error(req, err);
         }
+
+        if (self.options.cache && self.options.cache.page && self.options.cache.page.maxAge) {
+          self.setMaxAge(req, self.options.cache.page.maxAge);
+          self.sendETag(req);
+
+          if (self.doesETagMatch(req)) {
+            // Stop there and send a 304 status code; the cached response will be used
+            return res.sendStatus(304);
+          }
```

Comment:
FIXME: max-age isn't taken into account, the response is still cached even after the period of time set in max-age

Reply:
I am not sure I understand. Are you saying we're setting the header but the browser doesn't respect it? Are we setting the header correctly?

Reply:
It does but after the max-age period, it will just stop loading the cached content from the disk, and make the actual request.

But since Etag still matches (no modification has been made), the response status code 
will be `304`, and the content won't be fresh.

I might be missing an obvious Etag/Cache-Control mechanism here